Sources/Linq2DynamoDb.DataContext/TableDefinitionWrapper.cs

```diff
@@ -360,45 +369,120 @@ protected override void InitReader(ISupervisableEnumerable reader)
         /// <summary>
         /// Fills in and executes an update batch
         /// </summary>
-        private Task ExecuteUpdateBatchAsync(IDictionary<EntityKey, Document> addedEntities, IDictionary<EntityKey, Document> modifiedEntities, ICollection<EntityKey> removedEntities)
+        private Task ExecuteUpdateBatchAsync(IDictionary<EntityKey, Document> addedEntities, IDictionary<EntityKey, Document> modifiedEntities, IDictionary<EntityKey, Document> removedEntities)
         {
-            var batch = this.TableDefinition.CreateBatchWrite();
+            var dynamoDbUpdateTasks = AddEntriesAsync(addedEntities)
```

Comment:
I thought, we agreed on switching between two approaches (batches vs. individual tasks), depending on whether a Version field was defined or not...

Batches are much faster, so I wouldn't just get rid of them in favor of optimistic updates.